### ISSUE TITLE
How to specify the learning rate decay strategy during training? How to achieve it?

### ISSUE THREAD

**Issue by Wanghe1997** [ultralytics/yolov5#5715]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

For YOLOv5-v6.0
Such as StepLR，ExpLR，MultiStepLR，CosineLR and so on. In YOLOv5-v6.0，which learning rate decay strategy is used by default? If I want to change, which lines of code in which file should I change? thanks！

**Comment by glenn-jocher** [comment]
@Wanghe1997 YOLOv5 uses cosine LR scheduler by default:
https://github.com/ultralytics/yolov5/blob/8df64a912274ea3a82df2f96f0e3c3ab95713502/train.py#L167-L172

<img width="850" alt="LR Curves" src="https://user-images.githubusercontent.com/26833433/119332195-8e113a80-bc88-11eb-8eb7-7494ff1d5c01.png">

**Comment by Wanghe1997** [comment]
> @Wanghe1997 YOLOv5 uses cosine LR scheduler by default:
> 
> https://github.com/ultralytics/yolov5/blob/8df64a912274ea3a82df2f96f0e3c3ab95713502/train.py#L167-L172
> 
> <img alt="LR Curves" width="850" src="https://user-images.githubusercontent.com/26833433/119332195-8e113a80-bc88-11eb-8eb7-7494ff1d5c01.png">

In addition to cosine LR scheduler, are there other strategies written in the program?

**Comment by glenn-jocher** [comment]
@Wanghe1997 linear is written in with python train.py --linear, but you can also use any other custom scheduler you want with a lambda function or any premade schedulers here:
https://pytorch.org/docs/stable/optim.html

<img width="915" alt="Screenshot 2021-11-19 at 13 49 45" src="https://user-images.githubusercontent.com/26833433/142625546-2e54c09e-5819-4b07-a961-f6265bba43f5.png">

**Comment by Wanghe1997** [comment]
> @Wanghe1997 linear is written in with python train.py --linear, but you can also use any other custom scheduler you want with a lambda function or any premade schedulers here: https://pytorch.org/docs/stable/optim.html
> 
> <img alt="Screenshot 2021-11-19 at 13 49 45" width="915" src="https://user-images.githubusercontent.com/26833433/142625546-2e54c09e-5819-4b07-a961-f6265bba43f5.png">

Thank you for your reply. Can you answer my supplementary question in another issue? 
https://github.com/ultralytics/yolov5/issues/5653#issuecomment-973667987

**Comment by adityatandon** [comment]
For anyone looking for an updated answer in 2022, from YOLOv5-version 6.1 onwards, the default learning rate scheduler has been changed to linear instead of the cosine lr scheduler.

**Comment by Sylus1000** [comment]
Scheduler code in in line 158 of train.py. Its either LamdaLR or OneCylceLR (cosine)
[https://github.com/ultralytics/yolov5/blob/064365d8683fd002e9ad789c1e91fa3d021b44f0/train.py#L158-L163](url)

**Comment by glenn-jocher** [comment]
@adityatandon thank you for the update! The YOLOv5 team is constantly improving the framework based on the latest research and user feedback. If you have any further questions or need assistance with anything else, feel free to ask here.
